src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -2,18 +2,22 @@
 {-# LANGUAGE TypeApplications     #-}
 {-# LANGUAGE TypeOperators        #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
+{-# OPTIONS_GHC -Wno-unused-matches #-}
```

Comment:
This will be removed after specializing the `arbitrary` instance for `NonInteractiveProof`.